source/merge-instances.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+const merge = require('./merge');
+
+module.exports = (instances, methods) => {
+	for (const [index, instance] of instances.entries()) {
+		if (Reflect.has(instance.defaults, 'mergeable') && !instance.defaults.mergeable) {
+			throw new Error(`Instance ${index} is not mergeable.`);
```

Comment:
Note: many errors should have their own class.

Reply:
I think the error should say *why* it's not mergeable. That's not clear.

Reply:
It's not mergeable because `instance.defaults.mergeable` is set to `false`. This is the only way to make it unmergeable.

> I think the error should say why it's not mergeable.

`Instance {$index} refused to be merged`?

Reply:
The `mergeable` option is not documented and it's not clear what use-case it serves? Why do you need to make an instance unmergable?

Reply:
> The mergeable option is not documented (I we decide to keep it, it should be)

Yeah, it should be.

> Why do you need to make an instance unmergable?

I assume instances can be very complicated, e.g. override the current behaviour, or merging it with other instances may somehow result in incompatibility. I see no other reasons. If you feel it's not needed, I'll remove it.

Reply:
I'll remove it. As for now I don't see any useful use-case. We can add this feature later if needed.